app/models/manageiq/providers/hawkular/common/event_catcher/runner_mixin.rb

```diff
@@ -0,0 +1,50 @@
+module ManageIQ::Providers::Hawkular::Common::EventCatcher::RunnerMixin
+  extend ActiveSupport::Concern
+
+  def log_handle
+    self.class.log_handle
+  end
+
+  def reset_event_monitor_handle
+    @event_monitor_handle = nil
+  end
+
+  def stop_event_monitor
+    @event_monitor_handle.try(:stop)
+  ensure
+    reset_event_monitor_handle
+  end
+
+  def monitor_events
+    event_monitor_handle.start
+    event_monitor_handle.each_batch do |events|
+      event_monitor_running
+      new_events = events.select { |e| whitelist?(e) }
```

Comment:
@moolitayer The blacklist logic, if I remember correctly, is a built-in concept from MIQ.  Event types can be blacklisted, I think, due to potential flooding or other excessive behavior. I'm not not totally sure of the mechanism, perhaps @durandom can comment.  But the handling and naming was made to be like other providers.   The Whitelist was something I added for the hawkular provider's event catcher.  Whether the whitelist is necessary is debatable.   I thought it would be a good idea to keep unexpected event types from progressing downstream.  This was put in place before we actually used event tags to limit our queries.  If we can assume that the events fetched are already "white", due to the fact that they have already been tagged,  then there really may not be a need for the whitelist logic.  And because the whitelist currently requires an actual code change to support new entries, we should probably look hard at whether it is something we want to keep.